### ISSUE TITLE
Secrets leakage when describing failing resource

### ISSUE THREAD

**Issue by AtzeDeVries** [upbound/upjet#154]
### What happened?
While developing a new resource in a provider i got this when describing a failing resource i get return (next to the 'normal info')
```
> k describe repositoryfile sample-file.txt

Name:         sample-file.txt
Namespace:
Labels:       <none>
Annotations:  crossplane.io/external-name: sample-file.txt
API Version:  repo.github.upbound.io/v1alpha1
Kind:         RepositoryFile
Metadata:
  Creation Timestamp:  2023-02-03T09:24:31Z

... skipped info by me

Status:
  At Provider:
  Conditions:
    Last Transition Time:  2023-02-03T09:25:53Z
    Message:               connect failed: cannot get a terraform workspace for resource: cannot init workspace: There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.
╷
│ Error: Invalid resource name
│
│   on main.tf.json line 1, in resource.github_repository_file:
│    1: {"provider":{"github":{"owner":"**redacted**","token":"***redacted***"}},"resource":{"github_repository_file":{"sample-file.txt":{"branch":"add-file-branch","commit_author":"Crossplane Github Provider","commit_email":"github-provider@crossplane.com","commit_message":"managed by crossplane provider","content":"I am an crossplane \nprovider. This should be nice.\n","file":"sample-file.txt","lifecycle":{"prevent_destroy":true},"name":"sample-file.txt","repository":"github-crossplane-file-test"}}},"terraform":{"required_providers":{"github":{"source":"integrations/github","version":"5.15.0"}}}}
│
│ A name must start with a letter or underscore and may contain only letters,
│ digits, underscores, and dashes.
╵

: exit status 1
```

The issue here is that the full `main.tf.json` and this file as potentially secrets in there. 

### How can we reproduce it?

* Create a provider using upjet
* Create a resource which does not have the `name` input
* configure externalname with `config.NameAsIdentifier`
* Run the provider locally (`make run`) 
* Deploy the resource with an invalid nameing pattern (non dns spec)
* Describe the failing resource

### What environment did it happen in?
Crossplane version: docker.io/upbound/crossplane:v1.10.2-up.1

* Kubernetes version: 1.25
* Kind
* Mac OS 12.6.1

> initially opened an issue at crossplane here: https://github.com/crossplane/crossplane/issues/3721

**Comment by jeanduplessis** [comment]
Thanks, @AtzeDeVries we'll look into this and report back.

**Comment by jeanduplessis** [comment]
Due to the potential leakage of sensitive information, I will mark this issue as `severity:high` for now. We might downgrade this severity after first investigation.

**Comment by ulucinar** [comment]
Hi @AtzeDeVries,
Could you please confirm that you observed this issue with the async controller [configuration](https://github.com/upbound/upjet/blob/aeb5f15054f18d0eff52a75e71ec0f562fb79ffb/pkg/config/resource.go#L282) (which is the default as of now)? I.e., that you have not explicitly set `UseASync` to `false` as part of your resource configuration?

**Comment by AtzeDeVries** [comment]
Hi @ulucinar Have not touched that setting. Source code (excluding the error mentioned in this issue) is here https://github.com/coopnorge/provider-github

**Comment by AtzeDeVries** [comment]
@ulucinar I uploaded (and retested) the issue, check this draft pr: https://github.com/coopnorge/provider-github/pull/6

**Comment by ulucinar** [comment]
Thanks @AtzeDeVries, the draft was very helpful. Opened the PR #156 to address this immediate issue. Contents of `main.tf.json` appears in the `terraform init` output and the PR proposes to filter sensitive information from that output like we do in other sites.

**Comment by ulucinar** [comment]
I've reproduced the issue against the draft PR and validated the fix as mentioned in the description of PR #156.
